frontend/src/lib/components/apps/components/display/AppPdf.svelte

```diff
@@ -0,0 +1,294 @@
+<script lang="ts">
+	import { getContext } from 'svelte'
+	import { twMerge } from 'tailwind-merge'
+	import { getDocument, type PDFDocumentProxy, type PDFPageProxy } from 'pdfjs-dist'
+	import 'pdfjs-dist/build/pdf.worker.entry'
+	import type { AppInput } from '../../inputType'
+	import type { AppEditorContext, ComponentCustomCSS } from '../../types'
+	import { concatCustomCss } from '../../utils'
+	import InputValue from '../helpers/InputValue.svelte'
+	import { debounce, throttle } from '../../../../utils'
+	import { Button } from '../../../common'
+	import { Download, Loader2, ZoomIn, ZoomOut } from 'lucide-svelte'
+	import { fade } from 'svelte/transition'
+	import { findGridItem } from '../../editor/appUtils'
+
+	export let id: string
+	export let configuration: Record<string, AppInput>
+	export const staticOutputs: string[] = ['loading']
+	export let customCss: ComponentCustomCSS<'container'> | undefined = undefined
+
+	const { app, mode } = getContext<AppEditorContext>('AppEditorContext')
+	// const resizeObserver = new ResizeObserver(() => {
+	// 	console.log('debounced render')
+	// 	debouncedRender(true, true)
+	// })
+
+	let source: string | ArrayBuffer | undefined = undefined
+	let wrapper: HTMLDivElement | undefined = undefined
+	let error: string | undefined = undefined
+	let doc: PDFDocumentProxy | undefined = undefined
+	let pages: PDFPageProxy[] = []
+	let zoom = 100
+	let controlsWidth: number | undefined = undefined
+	let controlsHeight: number | undefined = undefined
+	let pageNumber = 1
+
+	$: gridItem = findGridItem($app, id)
```

Comment:
This reactive statement is not necessary, you can call findGridItem where you use it